src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2067,6 +2081,21 @@ public int getSSTableLevel()
         return sstableMetadata.sstableLevel;
     }
 
+    /**
+     * Mutate Stats Metadata with a lock to avoid racing with entire-sstable-streaming and then reload sstable metadata
+     *
+     * @param description on changed attributes
+     * @param transform function to mutate stats metadata
+     */
+    public void mutateAndReloadStats(String description, UnaryOperator<StatsMetadata> transform) throws IOException
```

Comment:
So I guess there's one other thing you could do that would again simplify the tracing stuff we added back, and that's breaking `mutateAndReloadStats()` back into a few different methods. (i.e. One for changing repairedAt time, one for level changes, etc.) It might at least make piecing together the actual trace message easier. The `runWithWriteLock()` wrapping isn't really that much boilerplate to duplicate among them for the sake of clarity.

Reply:
good idea.